sql_generators/mobile_kpi_support_metrics/__init__.py

```diff
@@ -27,70 +28,118 @@
-        "type": "STRING",
-        "description": "Play store source the profile is attributed to.",
-    },
-    {
-        "name": "meta_attribution_app",
-        "type": "STRING",
-        "description": "Facebook app linked to paid marketing.",
-    },
-    *INSTALL_SOURCE,
-    *ADJUST_FIELDS,
-]
-
-FIREFOX_IOS_ATTRIBUTION_FIELDS = [
-    {
-        "name": "is_suspicious_device_client",
-        "type": "BOOLEAN",
-        "description": "Flag to identify suspicious device users, see bug-1846554 for more info.",
-    },
-    *ADJUST_FIELDS,
-]
+
+class Pings(Enum):
+    """TODO."""
+
+    metrics: str = "metrics"
+    first_session: str = "first_session"
+    baseline: str = "baseline"
+
+
+@dataclass
+class AttributionFieldGroup:
+    """TODO."""
+
+    name: str
+    source_pings: list[Pings]
+    fields: list[dict[str, str]]
+
+
+class AttributionFields:
+    """TODO."""
+
+    install_source = AttributionFieldGroup(
+        name="install_source",
+        source_pings=[Pings.metrics],
+        fields=[
+            {
+                "name": "install_source",
+                "type": "STRING",
+                "description": "The source of a profile installation.",
+            },
+        ],
+    )
+    adjust = AttributionFieldGroup(
+        name="adjust",
+        source_pings=[Pings.metrics, Pings.first_session],
+        fields=[
+            {
+                "name": "adjust_ad_group",
+                "type": "STRING",
+                "description": "Adjust Ad Group the profile is attributed to.",
+            },
+            {
+                "name": "adjust_campaign",
+                "type": "STRING",
+                "description": "Adjust Campaign the profile is attributed to.",
+            },
+            {
+                "name": "adjust_creative",
+                "type": "STRING",
+                "description": "Adjust Creative the profile is attributed to.",
+            },
+            {
+                "name": "adjust_network",
+                "type": "STRING",
+                "description": "Adjust Network the profile is attributed to.",
+            },
+        ],
+    )
+    play_store = AttributionFieldGroup(
+        name="play_store",
+        source_pings=[Pings.first_session],
+        fields=[
+            {
+                "name": "play_store_attribution_campaign",
+                "type": "STRING",
+                "description": "Play store campaign the profile is attributed to.",
+            },
+            {
+                "name": "play_store_attribution_medium",
+                "type": "STRING",
+                "description": "Play store Medium the profile is attributed to.",
+            },
+            {
+                "name": "play_store_attribution_source",
+                "type": "STRING",
+                "description": "Play store source the profile is attributed to.",
+            },
+            # TODO: decide if this should be added here?
```

Comment:
@bani do you think we should include this here? It was added as part of trying to get a better understanding of the impact of DMA on fenix.

Reply:
There are over 100K distinct values just in the last 3 days, so no.
I know they can filter DMA choice screen from attribution_source. It seems this field has the full query string with all the parameters, but it's preferable to use the parsed ones.

Reply:
That makes sense, thanks. For now I will leave it out and in the next PR adding the attribution table I will tweak this to be only included in it to maintain the same attribution information as we currently have in the exiting version of the clients table.